src/test/java/com/bookappstore/service/impl/BookServiceImpTest.java

```diff
@@ -0,0 +1,282 @@
+
+        when(bookRepository.findBookById(idBook)).thenReturn(Optional.of(book));
+        when(bookMapper.toDto(book)).thenReturn(bookDto);
+
+        BookDto bookDtoServiceById = bookService.findById(idBook);
+
+        assertThat(bookDtoServiceById).isEqualTo(bookDto);
+        verify(bookRepository, times(1)).findBookById(idBook);
+        verify(bookMapper, times(1)).toDto(book);
+        verifyNoMoreInteractions(bookRepository, bookMapper);
+    }
+
+    @Test
+    @DisplayName("Using wrong id, method must throw an exception")
+    void getById_NonExistingId_ShouldReturnException() {
+        when(bookRepository.findBookById(anyLong())).thenReturn(Optional.empty());
+        Assertions.assertThrows(EntityNotFoundException.class,
+                () -> bookService.findById(anyLong()));
+    }
+
+    @Test
+    @DisplayName("Saving book with correct fields in a right way")
+    void save_NormalBook_ShouldReturnBook() {
+        CreateBookRequestDto testBookRequestDto = createTestBookRequestDto();
+        Book bookWithoutId = createTestBookWithoutId();
+        BookDto bookDto = createBookDto();
+
+        when(bookMapper.toModel(testBookRequestDto)).thenReturn(bookWithoutId);
+        when(bookRepository.save(bookWithoutId)).thenReturn(bookWithoutId);
+        when(bookMapper.toDto(bookWithoutId)).thenReturn(bookDto);
+
+        BookDto bookServiceById = bookService.save(testBookRequestDto);
+
+        assertThat(bookServiceById).isEqualTo(bookDto);
+    }
+
+    @Test
+    public void testFindAllBooksWithCategories() {
+
+        Book book1 = createTestBook();
+        book1.setId(1L);
+
+        Book book2 = createTestBook();
+        book2.setId(2L);
+
+        BookDto bookDto1 = createBookDto();
+        bookDto1.setTitle("Book 1");
+        bookDto1.setAuthor("Author 1");
+
+        BookDto bookDto2 = createBookDto();
+        bookDto2.setTitle("Book 2");
+        bookDto2.setAuthor("Author 2");
+
+        Pageable pageable = PageRequest.of(0, 10);
+        List<Book> books = List.of(book1, book2);
+        List<BookDto> bookDtos = List.of(bookDto1, bookDto2);
+
+        Mockito.when(bookRepository.findAllWithCategories(pageable)).thenReturn(books);
+        Mockito.when(bookMapper.toDtoList(books)).thenReturn(bookDtos);
+
+        List<BookDto> result = bookService.findAll(pageable);
+
+        assertNotNull(result);
+        assertEquals(bookDtos.size(), result.size());
+        verify(bookRepository).findAllWithCategories(pageable);
+        verify(bookMapper).toDtoList(books);
+    }
+
+    @Test
+    @DisplayName("Update book by Id")
+    public void updateBook_UpdateBookById_Correct() {
+        Set<Category> categories = Set.of(createCategory(1L, "Fiction"));
+        Book book = createTestBook();
+        CreateBookRequestDto createBookRequestDto = createTestBookRequestDto();
+        Mockito.when(categoryRepository.findByIdIn(createBookRequestDto.getCategoriesIds()))
+                .thenReturn(categories);
```

Comment:
Keep all mocks in one group, good practice to follow GWT grouping as well
// Given
... 
// When
...
// Than
...